apps/core-app/src/components/customFields/ShamanDeluxe.tsx

```diff
@@ -0,0 +1,75 @@
+import { ComponentProps, useMemo } from 'react';
+import styled from 'styled-components';
+import { RegisterOptions, useFormContext } from 'react-hook-form';
+
+import {
+  ignoreEmptyVal,
+  isNumberish,
+  SHAMAN_PERMISSIONS,
+} from '@daohaus/common-utilities';
+import { Buildable, DataSm, ShamanPermission } from '@daohaus/ui';
+
+import { useDao } from '../../contexts/DaoContext';
+
+const Secondary = styled.span`
+  color: ${(props) => props.theme.secondary};
+`;
+
+const DeluxeBox = styled.div`
+  p {
+    margin-bottom: 1.2rem;
+  }
+`;
+
+export const ShamanDeluxe = (
+  props: Buildable<
+    ComponentProps<typeof ShamanPermission> & { watchAddressField?: string }
+  >
+) => {
+  const { id = 'shamanPermission', watchAddressField = 'shamanAddress' } =
+    props;
+  const { watch } = useFormContext();
+  const { dao } = useDao();
+
+  const [shamanPermission, shamanAddress] = watch([id, watchAddressField]);
+
+  const oldShamanLevel = useMemo(() => {
+    if (!dao || !shamanPermission || !shamanAddress) return;
+
+    // Spelling error 'shamen' exists on the type definiton in Dao data.
+    // This is a temporary fix until the type definition is updated.
+    // Didin't want to break antything.
+    return dao?.shamen?.find((shaman) => shaman.shamanAddress === shamanAddress)
+      ?.permissions;
+  }, [dao, shamanPermission, shamanAddress]);
+
+  const newRules: RegisterOptions = {
+    validate: (val) =>
+      ignoreEmptyVal(val, (val) => {
+        if (oldShamanLevel == null) return true;
+        return Number(oldShamanLevel) > Number(val)
+          ? true
+          : 'Shaman permission level can only go down';
+      }),
+    ...props.rules,
```

Comment:
should this technically be above `validate` so it does not accidentally override validate in the rules?

Reply:
I've been experimenting with spreading the rules after the declaration so that wrapper components can easily overwrite validation rules if need be.